### PR TITLE
TRUNK-4872 : Added unit tests to ModuleUtil - getPackagesFromFile, fi…

### PR DESCRIPTION
**Pull request by Ruhanga**
Link to ticket:  https://issues.openmrs.org/browse/TRUNK-4872

### CODE REVIEW

**Comment by dkayiwa** [comment]
Do you happen to know why the test coverage decreased?

**Comment by Ruhanga** [comment]
> Do you happen to know why the test coverage decreased?

@dkayiwa , I have no Idea why the coverage bot commented this way for the lines I have not included not even related to what I was testing. I have pushed updates to see the outcome.

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java [hunk 1]
```diff
@@ -705,7 +708,52 @@ public void expandJar_shouldExpandFileWithParentTreeIfNameIsFileAndKeepFullPathI
...
+	@Test
+	public void file2url_shouldReturnNullIfFileIsNull() throws MalformedURLException {
+		URL nullURL = ModuleUtil.file2url(null);
+		Assert.assertNull(nullURL);
+	}
 
+	/**
+	* @see ModuleUtil#file2url(File)
+	*/
+	@Test(expected = MalformedURLException.class)
```
[the remark this answers is not in the document]
Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: @dkayiwa , if I got you correctly, yes for this particular test method. It's testing for the thrown exception. One of the condition for this is the `\0` (null) character in the file path `String` argument.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Your test says that file2url should throw a malformed exception.

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java [hunk 2]
```diff
@@ -705,7 +708,52 @@ public void expandJar_shouldExpandFileWithParentTreeIfNameIsFileAndKeepFullPathI
...
+		Assert.assertTrue(packageCollection.isEmpty());
+	}
+	
+	/**
+	* @see ModuleUtl#getPackagesFromFile(File)
+	*/
+	@Test
+	public void getPackagesFromFile_shouldSkipOptionalFoldersIfJarFile() throws IOException{
+		File f = new File(this.getClass().getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		File d = new File("/tmp/test1-1.0-SNAPSHOT.jar");
```
[the remark this answers is not in the document]
Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: I am running on Windows and I guess this has nothing to do with the raw java code unless I have mistakenly used some syntax/term tied to some OS.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Is windows happy with the /tmp/ path?

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Well caught. Thank you @dkayiwa , let me correct the English I used.

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Win 10 is happy. For the sake of lower version I will change this. Thank you.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Do lower versions complain?

[the remark this answers is not in the document]
Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: They (lower versions of windows) do not complain. And with some googling, `tmp` or `temp` is not a reserved/forbidden file/folder name. This is same for other OSs so I am just going to leave this unchanged.

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java [hunk 3]
```diff
@@ -705,7 +708,52 @@ public void expandJar_shouldExpandFileWithParentTreeIfNameIsFileAndKeepFullPathI
...
+		FileUtils.copyFile(f, d);
+		Collection<String> packageCollection = ModuleUtil.getPackagesFromFile(d);
+		
+		Assert.assertFalse(packageCollection.isEmpty());
+		for (String string : packageCollection) {
+			Assert.assertFalse(string.contains("lib"));
+			Assert.assertFalse(string.contains("META-INF"));
+			Assert.assertFalse(string.contains("web.module"));
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Can you point me to the line where ModuleUtil deals with a folder named "web.module"?

Comment (Ruhanga) [reply api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Thank you @dkayiwa for this catch, it was a typo issue. But here is [the line](https://github.com/openmrs/openmrs-core/blob/4a0feb8da351088f25fdc4e6d324a1f277aa3410/api/src/main/java/org/openmrs/module/ModuleUtil.java#L1141) nearly related to what you asked. I am refactoring this to correct my mistake.

**Pull request merged**